test/parallel/test-async-wrap-constructor.js

```diff
@@ -0,0 +1,12 @@
+'use strict';
+require('../common');
+const assert = require('assert');
```

Comment:
Could you add a couple of lines describing what this is testing
https://github.com/nodejs/node/blob/master/doc/guides/writing-tests.md